app/views/users/new_association.erb

```diff
@@ -0,0 +1,60 @@
+<% content_for :head do %>
+  <%= javascript_include_tag "user" %>
+<% end %>
+
+<% content_for :heading_class, "pb-0" %>
+<% content_for :heading do %>
+  <div class='header-illustration new-user-main'>
+    <h1><%= t ".title" %></h1>
+  </div>
+  <div class='header-illustration new-user-arm d-none d-md-block'></div>
+<% end %>
+
+<div class="auth-container">
+  <div class="text-muted col-sm form-container">
+
+    <h4><%= t ".welcome" %></h4>
+
+    <%= bootstrap_form_for current_user, :url => { :action => "create_association" } do |f| %>
+      <%= hidden_field_tag("referer", h(@referer)) unless @referer.nil? %>
+      <%= f.hidden_field :auth_provider %>
+      <%= f.hidden_field :auth_uid %>
+
+      <% if current_user.errors[:email].empty? %>
```

Comment:
The reason for initially not displaying the email field is to simplify the screen and minimize the possibility of users being tempted to modify the value.

The field when there was no auth provider email available, or there already is an OSM user with the same email address.

Reply:
@tomhughes This is actually a bit tricky and probably rare scenario. User selects "login with X" and after logging in with X, X reports that user has an email address `user@X.com`. But that address is already used, likely by the same user. The implementation currently un-hides the email field and forces user to choose different email.

This is not really approprirate, since the user may not have another email address to offer. User would already have OSM username/password account. Perhaps the proper response would be to offer them to link their existing OSM account with auth_provider they chose.

Perhaps we could simply redirect them to "Login" page instead of offering them (and forcing them) to provide alternate email?

Reply:
I like it a lot when a service just connects both accounts and logs me in. One can un-link them later, manually. But the user intend was to login and use the social provider, so doing both and connecting the account follows this intend. I don't thing the app need to request user input on this.

Reply:
Hm, interesting. I thought that it may be a good idea to verify user's password as it is known to OSM. But maybe it is a valid assumption, at least when we trust 3rd party platform about the email address.

What do we do if we don't trust the email address (github, wikipedia, raw openid)? This would mean that user can create an account on github with someone elses email, then use that to login to OSM. Assuming github does not verify emails, this would allow hijacking OSM account...

Reply:
Well if we don't trust the email address then we do a confirmation on it anyway before the account is activated.

Reply:
I have updated the PR to modify the existing OSM account with `auth_uid` and `auth_provider` when the email reported by auth provider already has an OSM account. The existing `display_name` is not updated from auth_provider, the existing OSM display name is preserved.